### PR TITLE
Mocked signer api backend

### PR DESCRIPTION
**Pull request by jferrant**
The README has been updated to demonstrate how to use Swagger-UI, A server with simulated data in the database, and a server with just nothing in it. I have added Utoipa to all new commands and it has nicely generated an openapi.json file which can be used with a command like so: `openapi-generator generate -g rust-server -i ~/git/core-eng/stacks-signer-api/generated/openapi.json -o ./stacks-signer-api/generated/generated_code --minimal-update`

For review of the api structures more so than just the code, I recommend utilizing the swagger UI command to get a good overview of the transaction/vote routes and schemas that are there so far. 

run the following command in your terminal `stacks-signer-api swagger` and then navigate to `http://0.0.0.0:3030/swagger-ui/` in your browser.

NOTES: THIS IS NOT THE FINALIZED API
NEXT STEPS INCLUDE BUT ARE NOT LIMITED TO:
Add Utoipa to the signer/key mangament/configuration management (hence why I have no deleted the old README which contains documentation for the signer and key routes as they are now). 
Still need to add digital signature authentication of API calls
The database is built of finalized response/request structs rather than what makes functional sense from the binary perspective.
The signer needs to be linked to the transactions available to it
Configuration routes/db need to be added

Closes https://github.com/Trust-Machines/stacks-sbtc/issues/475
Tests will be added in a sep PR: https://github.com/Trust-Machines/stacks-sbtc/issues/476

### CODE REVIEW

File: stacks-signer-api/outdated-README.md
[the remark this answers is not in the document]
Comment (netrome) [reply stacks-signer-api/outdated-README.md:1]: Nevermind, I see why you keep it around. Perhaps it's worth adding a sentence in the beginning of the file to explain why it's still alive.

File: stacks-signer-api/src/db/transaction.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+use crate::transaction::Transaction;
+use sqlx::{Row, SqlitePool};
+
+// SQL queries used for performing various operations on the "transactions" table.
+const SQL_INSERT_TRANSACTION: &str = r#"INSERT INTO transactions (
+    txid, transaction_kind, transaction_block_height, transaction_deadline_block_height,
+    transaction_amount, transaction_fees, memo, transaction_originator_address,
+    transaction_debit_address, transaction_credit_address
+) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"#;
```
Comment (netrome) [inline stacks-signer-api/src/db/transaction.rs:16]: Nit: Some formatting would be nice to have here

File: stacks-signer-api/src/main.rs [hunk 2]
```diff
@@ -16,8 +48,140 @@ pub fn initiate_tracing_subscriber() {
...
+        .expect("Failed to parse provided address and port into socket address");
+    println!("Serving warp server on {}", server);
+    warp::serve(routes).run(server).await;
+}
+
+async fn run_swagger(pool: SqlitePool, args: SwaggerArgs) {
+    let config = Arc::new(Config::from("/api-doc.json"));
+    let api_doc = warp::path("api-doc.json")
+        .and(warp::get())
+        .map(|| warp::reply::json(&ApiDoc::openapi()));
```
Comment (netrome) [inline stacks-signer-api/src/main.rs:126]: Nit: It would be nice to have this doc path configurable as a CLI option

**Review by netrome — COMMENTED** [review]
So far so good. Nothing major from my side except a few commented out lines which I assume should be handled before this is ready to merge.

File: stacks-signer-api/src/db/transaction.rs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+) -> Result<(), DatabaseError> {
+    let transaction_kind = &transaction.transaction_kind.to_string();
+    let transaction_block_height = transaction.transaction_block_height;
+    let transaction_deadline_block_height = transaction.transaction_deadline_block_height;
+    let transaction_amount = transaction.transaction_amount;
+    let transaction_fees = transaction.transaction_fees;
+    let memo = &transaction.memo;
+    let transaction_originator_address = ""; //&transaction.transaction_originator_address.0;
+    let transaction_debit_address = ""; //&transaction.transaction_debit_address.0;
+    let transaction_credit_address = ""; //&transaction.transaction_credit_address.0;
```
[the remark this answers is not in the document]
Comment (jferrant) [reply stacks-signer-api/src/db/transaction.rs:48]: No it is ready to merge with it like that. But yes it needs updating. I have created a ticket: https://github.com/Trust-Machines/stacks-sbtc/issues/472

File: stacks-signer-api/generated/openapi.json
[the remark this answers is not in the document]
Comment (jferrant) [reply stacks-signer-api/generated/openapi.json:1]: Will do :D

File: stacks-signer-api/outdated-README.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
+# Stacks Signer API
+TODO: Remove this README when Utoipa is added to the signer and key management
```
Comment (netrome) [inline stacks-signer-api/outdated-README.md:2]: You'll get bonus points here if you create an issue and link it in the TODO.

File: stacks-signer-api/src/db/transaction.rs [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    sqlx::query(SQL_INSERT_TRANSACTION)
```
[the remark this answers is not in the document]
Comment (netrome) [reply stacks-signer-api/src/db/transaction.rs:37]: The sqlx PR is merged now, so I guess it's time for another rebase?

**Review by netrome — APPROVED** [review]

File: stacks-signer-api/outdated-README.md (hunk 4 above)
Comment (jferrant) [reply stacks-signer-api/outdated-README.md:2]: Done :D https://github.com/Trust-Machines/stacks-sbtc/issues/420

Comment (netrome) [reply stacks-signer-api/outdated-README.md:2]: Yay! ⭐⭐⭐⭐⭐

**Pull request merged**